test/images/agnhost/README.md

```diff
@@ -72,6 +72,121 @@ created with the `pause` argument instead, allowing us execute multiple commands
+  `417 Expectation Failed` if it did not, or `400 Bad Request` if any of the endpoint's parameters
+  is invalid. The endpoint's parameters are:
+  - `host`: The host that will be dialed.
+  - `port`: The port that will be dialed.
+  - `request`: The HTTP endpoint or data to be sent through UDP. If not specified, it will result
+    in a `400 Bad Request` status code being returned.
+  - `protocol`: The protocol which will be used when making the request. Default value: `http`.
+    Acceptable values: `http`, `udp`.
+  - `tries`: The number of times the request will be performed. Default value: `1`.
+- `/echo`: Returns the given `msg` (`/echo?msg=echoed_msg`)
+- `/exit`: Closes the server with the given code (`/exit?code=some-code`). The `code`
+  is expected to be an integer [0-127] or empty; if it is not, it will return an error message.
+- `/healthz`: Returns `200 OK` if the server is ready, `412 Status Precondition Failed`
+  otherwise. The server is considered not ready if the UDP server did not start yet or
+  it exited.
+- `/hostname`: Returns the server's hostname.
+- `/hostName`: Returns the server's hostname.
+- `/shell`: Executes the given `shellCommand` or `cmd` (`/shell?cmd=some-command`) and
+  returns a JSON containing the fields `output` (command's output) and `error` (command's
+  error message). Returns `200 OK` if the command succeeded, `417 Expectation Failed` if not.
+- `/shutdown`: Closes the server with the exit code 0.
+- `/upload`: Accepts a file to be uploaded, writing it in the `/uploads` folder on the host.
+  Returns a JSON with the fields `output` (containing the file's name on the server) and
+  `error` containing any potential server side errors.
+
+Usage:
+
+```console
+    kubectl exec test-agnhost -- /agnhost netexec [--http-port <http-port>] [--udp-port <udp-port>]
+```
+
+### nettest
+
+A tiny web server for checking networking connectivity.
+
+Will dial out to, and expect to hear from, every pod that is a member of the service
+passed in the flag `--service`.
+
+Will serve a webserver on given `--port`, and will create the following endpoints:
+
+- `/read`: to see the current state, or `/quit` to shut down.
+
+- `/status`: to see `pass/running/fail` determination. (literally, it will return
+one of those words.)
+
+- `/write`: is used by other network test pods to register connectivity.
+
+Usage:
+
+```console
+    kubectl exec test-agnhost -- /agnhost nettest [--port <port>] [--peers <peers>] [--service <service>] [--namespace <namespace>] [--delay-shutdown <delay>]
+```
+
+### webhook (Kubernetes External Admission Webhook)
+
+The subcommand tests MutatingAdmissionWebhook and ValidatingAdmissionWebhook. After deploying
+it to kubernetes cluster, administrator needs to create a ValidatingWebhookConfiguration
+in kubernetes cluster to register remote webhook admission controllers.
+
+TODO: add the reference when the document for admission webhook v1beta1 API is done.
```

Comment:
nit: what doc? there are api reference docs, eg:
- https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#mutatingwebhookconfiguration-v1beta1-admissionregistration-k8s-io
- https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#validatingwebhookconfiguration-v1beta1-admissionregistration-k8s-io

Reply:
Since webhook already had a README, I just copied it over: https://github.com/kubernetes/kubernetes/pull/78390/files#diff-44438b0afef73ca4f8ed3f3f81eb0fe5L7 . We could add those 2 links in, if they are the doc references we need.

Reply:
I would suggest we update the docs, doesn't have to be in this PR if you'd rather keep it copy-paste

Reply:
Added.